engine/src/core/graphics/rendering/buffers/index_buffer.h

```diff
@@ -17,12 +17,12 @@ namespace Bald::Graphics {
         /**
         * @fn Create
         * @brief Creates data buffer based on currently chosen rendering API.
-        * @param [unsigned short*] data -> Pointer to the memory containing data (eg. triangle indices)
-        * @param [unsigned] count -> Number of data points.
+        * @param [unsigned*] data -> Pointer to the memory containing data (eg. triangle indices)
+        * @param [unsigned] size -> Size of data array in bytes.
         * @return [IndexBuffer*] Pointer to the index buffer object.
         */
 
-        static IndexBuffer* Create(unsigned short* data, unsigned count);
+        static IndexBuffer* Create(unsigned* data, unsigned size);
 
```

Comment:
What about code like that:
IndexBuffer* one = Create(args..);
IndexBuffer* two = Create(args...);
one = two;
IndexBuffer* other = new OpenGLIndexBuffer(one);
Right now it would delete buffer with the same index twice.
We have to make it invalid code or make it work properly. 
The same applies to VertexBuffer.

Reply:
In the given example, you just end up with a dangling pointer. I have changed it so that now we always return std::shared_ptr => in a case of dangling ptr the memory will be freed. When it comes to the second scenario, I think we should delete the copy ctor and operator=.